inject/src/main/java/io/micronaut/inject/annotation/MutableAnnotationMetadata.java

```diff
@@ -50,6 +50,15 @@ public class MutableAnnotationMetadata extends DefaultAnnotationMetadata {
     public MutableAnnotationMetadata() {
     }
 
+    MutableAnnotationMetadata(DefaultAnnotationMetadata annotationMetadata) {
```

Comment:
You should use existing `of` method that can properly handle defaults and repeatable propagation